Rubberduck.Core/UI/CodeExplorer/Commands/ReplaceProjectContentsFromFilesCommand.cs

```diff
@@ -16,21 +20,28 @@ public class ReplaceProjectContentsFromFilesCommand : ImportCommand
             IFileSystemBrowserFactory dialogFactory, 
             IVbeEvents vbeEvents, 
             IParseManager parseManager,
+            IDeclarationFinderProvider declarationFinderProvider,
+            IProjectsProvider projectsProvider,
+            IModuleNameFromFileExtractor moduleNameFromFileExtractor,
+            IEnumerable<IRequiredBinaryFilesFromFileNameExtractor> binaryFileExtractors,
+            IFileExistenceChecker fileExistenceChecker,
             IMessageBox messageBox) 
-            :base(vbe, dialogFactory, vbeEvents, parseManager, messageBox)
-        { }
+            :base(vbe, dialogFactory, vbeEvents, parseManager, declarationFinderProvider, projectsProvider, moduleNameFromFileExtractor, binaryFileExtractors, fileExistenceChecker, messageBox)
+        {}
 
         protected override string DialogsTitle => RubberduckUI.ReplaceProjectContentsFromFilesCommand_DialogCaption;
 
-        protected override void ImportFiles(ICollection<string> filesToImport, IVBProject targetProject)
+        protected override ICollection<QualifiedModuleName> ModulesToRemoveBeforeImport(IDictionary<string, QualifiedModuleName> existingModules)
         {
-            if (!UserConfirmsToReplaceProjectContents(targetProject))
-            {
-                return;
-            }
+            return DeclarationFinderProvider.DeclarationFinder
+                .UserDeclarations(DeclarationType.Module)
+                .Select(decl => decl.QualifiedModuleName)
+                .ToHashSet();
+        }
 
-            RemoveReimportableComponents(targetProject);
-            base.ImportFiles(filesToImport, targetProject);
+        protected override bool UserDeniesExecution(IVBProject targetProject)
```

Comment:
The usage of `Denies` doesn't flow here. Did you mean `Cancelled` or maybe `Declines`?

Reply:
I'll rename to _Declines_ then.